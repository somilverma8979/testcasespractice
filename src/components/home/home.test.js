import React from "react";
// import { render, screen, act } from "@testing-library/react"
import axios from 'axios'
import Home from "./home"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
const  {act} = renderer

it("loads the data on mount", async () => {
    const FAKE_HITS = [
        {
            "userId": 1,
            "id": 1,
            "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
            "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
        },
        {
            "userId": 1,
            "id": 2,
            "title": "qui est esse",
            "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
        }
    ];
    const axiosGetSpy = jest.spyOn(axios, 'get').mockResolvedValueOnce({ data: FAKE_HITS });// spy on axios and pass data we want in return 

    let component;
    await act(async () => {
        component = renderer.create(<Home/>); // create component snapshot we want to call axios in 
    });
    expect(axiosGetSpy).toBeCalledWith('https://jsonplaceholder.typicode.com/posts');// specify data or url in this case we want to pass
    expect(component.toJSON()).toMatchSnapshot();
    axiosGetSpy.mockRestore(); //  resets everything, which includes usage data, implementation and mock name.
})

// mock axios call , we make an mock array that we want when the call gets completed