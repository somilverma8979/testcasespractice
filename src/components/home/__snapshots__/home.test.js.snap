// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads the data on mount 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3"
  >
    <div
      className="mt-3 mb-3"
      style={
        Object {
          "border": "1px solid #000",
          "width": "18rem",
        }
      }
    >
      <div
        className="card-body"
      >
        Title: 
        <h5
          className="card-title"
        >
          sunt aut facere repellat provident occaecati excepturi optio reprehenderit
        </h5>
        Body:  
        <p
          className="card-text"
          style={
            Object {
              "textAlign": "justify",
            }
          }
        >
          quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <div
      className="mt-3 mb-3"
      style={
        Object {
          "border": "1px solid #000",
          "width": "18rem",
        }
      }
    >
      <div
        className="card-body"
      >
        Title: 
        <h5
          className="card-title"
        >
          qui est esse
        </h5>
        Body:  
        <p
          className="card-text"
          style={
            Object {
              "textAlign": "justify",
            }
          }
        >
          est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla
        </p>
      </div>
    </div>
  </div>
</div>
`;
